File: pipeline/src/main/java/com/ibrahimatay/stage/OutputMessagesStage.java
package com.ibrahimatay.stage;

import com.ibrahimatay.model.Message;
import com.ibrahimatay.model.Messages;
import com.ibrahimatay.stage.Stage;

public class OutputMessagesStage implements Stage<Messages> {
    @Override
    public Messages execute(Messages input) {
        for (Message message: input.messages()){
            System.out.println(message);
        }

        return input;
    }
}

File: DesignPatterns.Pipeline/Stage/OutputMessagesStage.cs
using DesignPatterns.Pipeline.Model;

namespace DesignPatterns.Pipeline.Stage;

public class OutputMessagesStage : IStage<Messages>
{
    public Messages Execute(Messages input)
    {
        input.MessageList.ForEach(Console.WriteLine);
        
        return input;
    }
}